src/status_im/utils/gfycat/core.cljs

```diff
@@ -12,3 +13,14 @@
     (native-module/generate-gfycat public-key)))
 
 (def generate-gfy (memoize build-gfy))
+
+(re-frame/reg-fx
+ :insert-gfycat-fx
+ (fn [[public-key path]]
+   (native-module/generate-gfycat-async public-key #(re-frame/dispatch [:set-in path %]))))
```

Comment:
Unless necessary, we generally avoid using `set-in` , but rather have a well defined event for this (i.e `gfycat-generated`), same for identicon